ticketutil/bugzilla.py

```
@@ -0,0 +1,367 @@
+
+    def _create_requests_session(self):
+        """
+        Returns to the super class if the authentication method is kerberos.
+        Creates a Requests Session and authenticates to base API URL with authentication other then kerberos.
+        We're using a Session to persist cookies across all requests made from the Session instance.
+        :return s: Requests Session.
+        """
+        if self.auth == 'kerberos':
+            return super(BugzillaTicket, self)._create_requests_session()
+
+        # If basic authentication is passed in, generate a token to be used in all requests.
+        try:
+            s = requests.Session()
+            r = s.get(self.auth_url, params=self.credentials, verify=False)
+            r.raise_for_status()
+            resp = r.json()
+            self.token = resp['token']
+            logging.debug("Create requests session: Status Code: {0}".format(r.status_code))
+            logging.info("Successfully authenticated to {0}.".format(self.ticketing_tool))
+            return s
+
+        # We log an error if authentication was not successful, because rest of the HTTP requests will not succeed.
+        # If authentication wasn't successful, a token will not be in resp. Add KeyError as exception.
+        except (KeyError, requests.RequestException) as e:
+            logging.error("Error authenticating to {0}. No valid credentials were provided.".format(self.auth_url))
+            logging.error(e.args[0])
+
+    def create(self, summary, description, **kwargs):
+        """
+        Creates a ticket.
+        The required parameters for ticket creation are summary and description.
+        Keyword arguments are used for other ticket fields.
+        :param summary: The ticket summary.
+        :param description: The ticket description.
+        :return
+        """
+        if summary is None:
+            logging.error("summary is a necessary parameter for ticket creation.")
+            return
```

Going alone with the comment above regarding returning: I feel like this return, while functionally ok, could be used to return some indication of failure, instead of relying solely on a logging message.